pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -81,25 +127,18 @@ func (c *Controller) onDefaultNetworkChange() {
 	c.reloadNetworkGateways()
 }
 
-// reloadNetworkLookup refreshes the meshNetworks configuration, network for each endpoint, and
-// recomputes network gateways.
-func (c *Controller) reloadNetworkLookup() {
-	c.reloadMeshNetworks()
-	c.onDefaultNetworkChange()
-}
-
 // reloadMeshNetworks will read the mesh networks configuration to setup
 // fromRegistry and cidr based network lookups for this registry
 func (c *Controller) reloadMeshNetworks() {
-	c.Lock()
-	defer c.Unlock()
-	c.networkForRegistry = ""
+	c.networkManager.Lock()
```

Comment:
Should this just be a method on NetworkManager? And should `c.networkFromMeshConfig` be read from `NetworkManager`?

Reply:
Or move the locking to some `Set` method at the end. Feels weird managing the lock from here.

Reply:
+1. I had similar doubt for extractGatewaysInner. It would be good to move locks to networkManager

Reply:
SGTM